File: frontend/src/app/models/workshop/job-card/job-card-info.ts
import { User } from '../../general/user';
import { VehicleInfo } from '../../vehicle/vehicle-info';

export class JobCardInfo {
    public uuid: string;
    public kmHour: string;
    public fleetNumber: string;
    public currentKmHour: string;
    public requestNumber: string;
    public vehicleTypeName: string;
    public jobCardStartDate: string;
    public jobCardEndDate: string;
    public operatorUser: User;
    public mechanicUser: User;
    public foremanUser: User;
    public supervisorUser: User;
    public jobCardStatus: string;
    public riskAssessment: boolean;
    public lockOutProcedure: boolean;
    public wheelNuts: boolean;
    public oilLevel: boolean;
    public machineGrease: boolean;
    public vehicleInfo: VehicleInfo;
    public description: string;

}

File: backend/src/main/java/com/minetec/backend/dto/info/workshop/JobCardInfo.java
package com.minetec.backend.dto.info.workshop;

import com.minetec.backend.dto.info.BasicUserInfo;
import lombok.Data;

import java.util.UUID;

@Data
public class JobCardInfo {

    private UUID uuid;
    private String fleetNumber;
    private String kmHour;
    private String currentKmHour;
    private String requestNumber;
    private String description;
    private String vehicleTypeName;
    private String jobCardStartDate;
    private String jobCardEndDate;
    private String jobCardStatus;
    private BasicUserInfo supervisorUser;
    private BasicUserInfo operatorUser;
    private BasicUserInfo mechanicUser;
    private BasicUserInfo foremanUser;
    private boolean riskAssessment;
    private boolean lockOutProcedure;
    private boolean wheelNuts;
    private boolean oilLevel;
    private boolean machineGrease;

}
